netmaster/objApi/apiController.go

```diff
@@ -0,0 +1,639 @@
+	return nil
+}
+
+// ServiceDelete deletes service
+func (ac *APIController) ServiceDelete(service *contivModel.Service) error {
+	log.Infof("Received ServiceDelete: %+v", service)
+	return nil
+}
+
+// ServiceInstanceCreate creates a service instance
+func (ac *APIController) ServiceInstanceCreate(serviceInstance *contivModel.ServiceInstance) error {
+	log.Infof("Received ServiceInstanceCreate: %+v", serviceInstance)
+	inst := serviceInstance
+
+	// Find the service
+	serviceKey := inst.TenantName + ":" + inst.AppName + ":" + inst.ServiceName
+	service := contivModel.FindService(serviceKey)
+	if service == nil {
+		log.Errorf("Service %s not found for instance: %+v", serviceKey, inst)
+		return errors.New("Service not found")
+	}
+
+	// Add links
+	modeldb.AddLinkSet(&service.LinkSets.Instances, inst)
+	modeldb.AddLink(&inst.Links.Service, service)
+
+	// setup links with volumes
+	for _, volumeName := range inst.Volumes {
+		// find the volume
+		volume := contivModel.FindVolume(inst.TenantName + ":" + volumeName)
+		if volume == nil {
+			log.Errorf("Could not find colume %s for service: %s", volumeName, inst.Key)
+			return errors.New("Could not find the volume")
+		}
+
+		// add Links
+		modeldb.AddLinkSet(&inst.LinkSets.Volumes, volume)
+		modeldb.AddLinkSet(&volume.LinkSets.ServiceInstances, inst)
+	}
+
+	return nil
+}
+
+// ServiceInstanceUpdate updates a service instance
+func (ac *APIController) ServiceInstanceUpdate(serviceInstance, params *contivModel.ServiceInstance) error {
+	log.Infof("Received ServiceInstanceUpdate: %+v, params: %+v", serviceInstance, params)
+	return nil
+}
+
+// ServiceInstanceDelete deletes a service instance
+func (ac *APIController) ServiceInstanceDelete(serviceInstance *contivModel.ServiceInstance) error {
+	log.Infof("Received ServiceInstanceDelete: %+v", serviceInstance)
+	return nil
+}
+
+// TenantCreate creates a tenant
+func (ac *APIController) TenantCreate(tenant *contivModel.Tenant) error {
+	log.Infof("Received TenantCreate: %+v", tenant)
+
+	if tenant.TenantName == "" {
+		return errors.New("Invalid tenant name")
+	}
+
+	// Get the state driver
+	stateDriver, err := utils.GetStateDriver()
+	if err != nil {
+		return err
+	}
+
+	// Build tenant config
+	tenantCfg := intent.ConfigTenant{
+		Name:           tenant.TenantName,
+		DefaultNetType: "vlan",
+		SubnetPool:     tenant.SubnetPool,
+		AllocSubnetLen: uint(tenant.SubnetLen),
+		VLANs:          tenant.Vlans,
+		VXLANs:         tenant.Vxlans,
+	}
+
+	// Create the tenant
+	err = master.CreateTenant(stateDriver, &tenantCfg)
+	if err != nil {
+		log.Errorf("Error creating tenant {%+v}. Err: %v", tenant, err)
+		return err
+	}
+
+	// Create private network for the tenant
+	err = contivModel.CreateNetwork(&contivModel.Network{
+		Key:         tenant.TenantName + ":" + "privateNet",
+		IsPublic:    false,
+		IsPrivate:   true,
+		Encap:       "vxlan",
+		Subnet:      "10.1.0.0/16",
```

Comment:
I am a bit wary about us always choosing these values by default inside netmaster as we might not always choose a good value. Should we have a way for user to provide these instead?